src/fheroes2/game/game_io.cpp

```diff
@@ -255,11 +264,13 @@ fheroes2::GameMode Game::Load( const std::string & filePath )
     }
 
     uint16_t endOfDataMarker = 0;
-    sb >> endOfDataMarker;
-
-    if ( sb.fail() || endOfDataMarker != SAV2ID3 ) {
-        DEBUG_LOG( DBG_GAME, DBG_WARN, "File " << filePath << " is corrupted" )
+    dataStream >> endOfDataMarker;
+    if ( dataStream.fail() ) {
+        showGenericErrorMessage();
+        return fheroes2::GameMode::CANCEL;
+    }
 
+    if ( endOfDataMarker != SAV2ID3 ) {
```

Comment:
Can we combine these two checks?

```suggestion
    if ( dataStream.fail() || ( endOfDataMarker != SAV2ID3 ) ) {
```

Reply:
It is done.